docs/mkdocs/catalogs.md

```diff
@@ -0,0 +1,235 @@
+
+# create a catalog from a pyiceberg catalog object
+_ = Catalog.from_iceberg(pyiceberg_catalog)
+
+# create a catalog from a unity catalog object
+_ = Catalog.from_unity(unity_catalog)
+
+# create a catalog from a pydict mapping names to tables
+catalog = Catalog.from_pydict({
+    "T": table_t,
+    "S": table_s,
+})
+
+# list available tables (for iceberg, the pattern is a prefix)
+catalog.list_tables(pattern=None)
+
+# get a table by name
+table_t = catalog.get_table("T")
+table_s = catalog.get_table("S")
+```
+
+**APIs**
+
+```python
+class Catalog(ABC):
+
+    def get_table(self, identifier: Identifier | str) -> Table: ...
+
+    def list_tables(self, pattern: str | None = None) -> list[str]: ...
+```
+
+**Notes**
+
+* You can create a Catalog from `pyiceberg` and `daft.unity` catalog objects.
+* We are actively working on additional catalog implementations.
+* Our next release will include DDL actions like `create_*` and `drop_*`.
+
+### Working with Tables
+
+The `Table` interface is a bridge from catalogs to dataframes. We can read tables into dataframes, and we can write dataframes to tables. You can work with a table independently of a catalog by using one of the factory methods, but it might not appear to provide that much utility over the existing `daft.read_` and `daft.write_` APIs. You would be correct in assuming that this is what is happening under the hood! The `Table` interface provides indirection over the table format itself and serves as a single abstraction for reading and writing that our catalogs can work with.
+
+**Examples**
+
+```python
+from daft import Table
+from pyiceberg.table import StaticTable
+
+# suppose you have a pyiceberg table
+pyiceberg_table = StaticTable("metadata.json")
+
+# we can make it a daft table to use daft's table APIS
+table = Table.from_iceberg(pyiceberg_table)
+
+# we can read a dataframe like `daft.read_iceberg(pyiceberg_table)`
+df = table.read()
+
+# you can also create temporary tables from dataframes
+daft.create_temp_table("my_temp_table", df.from_pydict({ ... }))
+
+# these will be resolved just like other tables
+df = daft.read_table("my_temp_table")
+```
+
+**APIs**
+
+```python
+class Table(ABC):
+
+    def read(self) -> DataFrame: ...
+
+    def show(self) -> None: ...
+
+    # coming soon!
+    # def append(self, df: DataFrame) -> None:  ...
+
+    # coming soon!
+    # def overwrite(self, df: DataFrame) -> None:  ...
+```
+
+**Notes**
+
+* Today you can read from `pyiceberg` and `daft.unity` table objects.
+* The daft catalog interface always returns tables.
+* We are actively working on additional table implementations and write APIs.
+
+### Working with Sessions
+
+Sessions hold attached objects, temporary objects, and stateful variables like the `current_catalog` and `current_namespace`.
+
+**Example**
+
+```python
+import daft
+
+# when you do `import daft`, there is an implicit global session
+sess = daft.current_session()
+
+# ...
+
+from daft import Session
```

Comment:
can we also have these docs/examples as part of the `Session` docstrings?

Reply:
Yes, definitely.